src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newArray = [];
+  let newObject = { ...state };
+
+  for (const oneAction of actions) {
+    if (oneAction.type === 'addProperties') {
+      const newObjectAdd = {
+        ...newObject, ...oneAction.extraData,
+      };
+
+      newArray.push(newObjectAdd);
+      newObject = { ...newObjectAdd };
+    } else if (oneAction.type === 'removeProperties') {
+      const newObjectDel = { ...newObject };
+
+      for (const arr of oneAction.keysToRemove) {
+        delete newObjectDel[arr];
+      }
+      newArray.push(newObjectDel);
+      newObject = { ...newObjectDel };
+    } else if (oneAction.type === 'clear') {
+      const clear = {};
+
+      newArray.push(clear);
+      newObject = {};
+    }
```

Comment:
use switch for such cases

Reply:
you can push new state object at the end of iteration 1 time instead of writing it into each if